subprojects/core/src/main/java/org/gradle/plugin/management/PluginResolutionSpec.java

```diff
@@ -0,0 +1,15 @@
+package org.gradle.plugin.management;
+
+import org.gradle.api.Incubating;
+
+/**
+ * This needs a better name.
+ *
+ * Allows user to specify details about the plugin that needs
+ * to be loaded.
+ */
+@Incubating
+public interface PluginResolutionSpec {
```

Comment:
This could be called `ConfigurablePluginRequest` and include things like `setVersion`.